.github/workflows/docs.yml

```diff
@@ -53,13 +56,17 @@ jobs:
         with:
           path: doc/deploy
 
-#   deploy:
-#     runs-on: ubuntu-22.04
-#     needs: build
-#     if: github.event_name != 'pull_request'
-#     permissions:
-#       pages: write
-#       id-token: write
-#     steps:
-#       - name: Deploy to GitHub Pages
-#         uses: actions/deploy-pages@v4
+  deploy:
+    runs-on: ubuntu-22.04
+    needs: build
+    if: github.event_name != 'pull_request'
```

Comment:
One minor thing- I think we can remove this line, because based on the docs for deploy-pages action: https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site this won't run on pull requests

Reply:
I tried it, and the deploy job failed in CI.